client/components/Editor/plugins/inputRules.ts

```diff
@@ -104,31 +104,54 @@ const inlineMathRule = (
 // textblock starting with two dollar signs into a math block.
 const blockMathRule = (nodeType) => makeBlockMathInputRule(REGEX_BLOCK_MATH_DOLLARS, nodeType);
 
-const HTTP_MAILTO_REGEX = new RegExp(
-	// eslint-disable-next-line no-useless-escape
-	/(?:(?:(https|http|ftp)+):\/\/)?(?:\S+(?::\S*)?(@))?(?:(?:([a-z0-9][a-z0-9\-]*)?[a-z0-9]+)(?:\.(?:[a-z0-9\-])*[a-z0-9]+)*(?:\.(?:[a-z]{2,})(:\d{1,5})?))(?:\/[^\s]*)?\s$/,
-);
+const emailOrUriRegexBase =
+	'(?<emailOrUri>(?:(?:(https|http|ftp)+)://)?(?:\\S+(?::\\S*)?(?<atSign>@))?(?:(?:([a-z0-9][a-z0-9-]*)?[a-z0-9]+)(?:\\.(?:[a-z0-9-])*[a-z0-9]+)*(?:\\.(?:[a-z]{2,})(:\\d{1,5})?))(?:/[^\\s]*)?)';
```

Comment:
I dropped the `\` in `[a-z0-9\-]` which is what eslint was complaining about. I checked myself that `[a-z0-9-]` unambiguously matches the `-` character.

Reply:
The other changes to the regex here (since it's hard to look at) are that I pulled the `\s` at the end out since I only want it some of the time, and I added a few named capture groups to make the code below more readable.